src/context_free.mli

```diff
@@ -150,3 +150,11 @@ class map_top_down :
        Generated_code_hook.t (* default: Generated_code_hook.nop *)
   -> Rule.t list
   -> Ast_traverse.map_with_expansion_context
+
+class map_top_down_with_errors :
```

Comment:
Why do we want to export a new `map_top_down_with_errors` instead of just changing the type of the existing class?

Reply:
I think the idea is to not introduce any breaking changes (hence the presence of a "raising" version and a "result" version for many functions modified in #329).

I think however that even though `map_top_down` is public, it is not meant to be used by anyone else than ppxlib's driver, so it would be ok to change its signature.

Reply:
The main concern I have is that the code is nontrivial and two copies of it might easily drift out of sync. Also it will be easier to see in review that the behavior is the same if it's just changes to the original.